File: test/llvm_test_code/linear_constant/global_08.cpp
int g = 1;

int baz(int c) { // clang-format off
  return g + c;
}

int bar(int b) {
  return baz(b + 1);
}

int foo(int a) {
  return bar(a + 1);
} // clang-format on

int main() {
  g += 1;
  int i = 0;
  i = foo(1);
  return 0;
}

File: test/llvm_swift_test_code/linear_constant/global_08.swift
var g = 1

@main
struct MyMain {

  static func baz (_ c: Int)-> Int {
    return g + c
  }

  static func bar (_ b: Int)-> Int {
    return baz(b + 1)
  }

  static func foo (_ a: Int)-> Int {
    return bar(a + 1)
  }

    static func main() {
      g += 1
      var i = 0
      i = foo(1)
    }
}
